pkg/util/restoptions/configgetter.go

```diff
@@ -1,119 +1,50 @@
 package restoptions
 
 import (
-	"sync"
-
-	"github.com/golang/glog"
+	"strconv"
 
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apiserver/pkg/registry/generic"
-	"k8s.io/apiserver/pkg/registry/generic/registry"
 	"k8s.io/apiserver/pkg/server/options"
+	apiserverstorage "k8s.io/apiserver/pkg/server/storage"
 	serverstorage "k8s.io/apiserver/pkg/server/storage"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
 
+	"github.com/openshift/origin/pkg/cmd/openshift-apiserver/openshiftapiserver/configprocessing"
 	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
-	kubernetes "github.com/openshift/origin/pkg/cmd/server/kubernetes/master"
 )
 
-// configRESTOptionsGetter provides RESTOptions based on a provided config
-type configRESTOptionsGetter struct {
-	masterOptions configapi.MasterConfig
-
-	restOptionsLock sync.Mutex
-	restOptionsMap  map[schema.GroupResource]generic.RESTOptions
-
-	storageFactory        serverstorage.StorageFactory
-	defaultResourceConfig *serverstorage.ResourceConfig
-
-	cacheEnabled     bool
-	defaultCacheSize int
-	cacheSizes       map[schema.GroupResource]int
-	quorumResources  map[schema.GroupResource]struct{}
-
-	deleteCollectionWorkers int
-	enableGarbageCollection bool
-}
-
 // NewConfigGetter returns a restoptions.Getter implemented using information from the provided master config.
 // TODO: this class should either not need to know about configapi.MasterConfig, or not be in pkg/util
-func NewConfigGetter(masterOptions configapi.MasterConfig, defaultResourceConfig *serverstorage.ResourceConfig, resourcePrefixOverrides map[schema.GroupResource]string, enforcedStorageVersions map[schema.GroupResource]schema.GroupVersion, quorumResources map[schema.GroupResource]struct{}) (Getter, error) {
-	apiserverOptions, err := kubernetes.BuildKubeAPIserverOptions(masterOptions)
-	if err != nil {
-		return nil, err
-	}
-	storageFactory, err := kubernetes.BuildStorageFactory(apiserverOptions, enforcedStorageVersions)
-	if err != nil {
-		return nil, err
-	}
-	storageFactory.DefaultResourcePrefixes = resourcePrefixOverrides
-	storageFactory.StorageConfig.Prefix = masterOptions.EtcdStorageConfig.OpenShiftStoragePrefix
-
-	// perform watch cache heuristic like upstream
-	if apiserverOptions.Etcd.EnableWatchCache {
-		glog.V(2).Infof("Initializing cache sizes based on %dMB limit", apiserverOptions.GenericServerRunOptions.TargetRAMMB)
```

Comment:
the log message is gone. Intentionally?

Reply:
> the log message is gone. Intentionally?

Yeah.  If we need to expose it, we need to expose it a different way.  Metric probably.